tests/test_exec.py

```diff
@@ -62,11 +64,71 @@ def test_exec_not_exists():
 
 def test_exec_detach_not_exists():
     return test_exec_not_exists_helper(False)
+
+NONKM_OK_CONF = "var/lib/krun/nonkm_exec_ok_config"
+
+def test_exec_non_km_payload_helper(execpath, nonkmok):
+    # only run these tests for krun
+    runtime = os.getenv("OCI_RUNTIME")
+    if os.path.basename(runtime) != "krun":
+        return 77  # skip this test if not using krun
+
+    conf = base_config()
+    conf['process']['args'] = ['/init', 'pause']
+    add_all_namespaces(conf)
+
+    # create allowed nonkm executables config file
+    nonkmok_file = "/tmp/nonkmok_config"
```

Comment:
I'd prefer these files to either be pre-creared, or if you need  to create it. use "here string" , something like 
```pythong
file_contenrt = """
# GENERATED FILE. DO NOT EDIT.
# 
line 1
line 2
"""
   open(target_file "w").write(file_content)
```

Reply:
The hash of the target program on each line needs to be created by the test program.
I'm not going to hard code hashes into the test.  Every time something changes related to the exec'ed file, we would need to compute a new hash and stick that into the here string.

Reply:
yes, you are right, I did not think about it.